src/views/template-creation/TemplateCreationView.tsx

```diff
@@ -1,51 +1,110 @@
 
-  const handleStepChange = (e: StepClickEvent) => {
-    setActiveStep(+e.currentTarget.getAttribute('data-idx'));
-  };
+  const basicInfo = useForm(stepsConfig[0].formConfig);
+  const githubConnection = useForm(stepsConfig[1].formConfig);
+  const technologiesOverview = useForm(stepsConfig[2].formConfig);
+
+  const formManagers = [basicInfo, githubConnection, technologiesOverview];
+
+  const changeStep = useCallback(
+    (stepValue: number, e?: FormSubmitEvent) => {
+      const getNextStep = () => {
+        const stepsDiff = stepValue - activeStep;
+
+        if (stepsDiff > 1) {
+          for (let i = activeStep + 1; i < formManagers.length - 1; i++) {
+            const submit = formManagers[i][3];
+
+            if (submit(e)) {
+              return i;
+            }
+          }
+        }
 
-  const goToNextStep = () => {
-    setActiveStep((prevActiveStep) => prevActiveStep + 1);
-  };
+        return stepValue;
+      };
 
-  const { label, description, formConfig } = templateCreationConfig[activeStep];
+      const submit = formManagers[activeStep][3];
 
-  const StepComponent = steps[activeStep];
+      if (stepValue > activeStep && submit(e)) {
+        return;
+      }
+
+      const nextStep = getNextStep();
+
+      if (nextStep > formManagers.length - 1) {
+        // AFTER BE IMPLEMENTATION ADD API CALL HERE
+        return;
+      }
+
+      setActiveStep(nextStep);
+    },
+    [activeStep, ...formManagers]
+  );
+
+  const onStepChange = useCallback(
+    (step: number) => {
+      changeStep(step);
+    },
+    [changeStep]
+  );
+
+  const onStepSubmit = useCallback(
+    (e: FormSubmitEvent) => {
+      changeStep(activeStep + 1, e);
+    },
+    [activeStep, changeStep]
+  );
+
+  const { label, description } = stepsConfig[activeStep];
+
+  const steps: Step[] = useMemo(
+    () =>
+      stepsConfig.map(
+        (c, idx) =>
+          ({
+            ...c,
+            status: getStepStatus(formManagers[idx]),
+            progress: getStepProgress(formManagers[idx])
+          } as Step)
+      ),
+    formManagers
+  );
 
   return (
     <div className={csx.templateCreationView}>
-      <h5 className={csx.stepTitle}>{label}</h5>
-      <span className={csx.stepDescription}>{description}</span>
-      <Steps
-        steps={templateCreationConfig}
-        activeStep={activeStep}
-        onStepClick={handleStepChange}
-      />
-      {StepComponent({
-        config: formConfig,
-        onSubmit: goToNextStep
-      })}
+      <Header description={description} label={label} />
+
+      <Steps steps={steps} onChange={onStepChange} />
+
+      {activeStep === 0 && <BasicInfoStep formManager={basicInfo} onSubmit={onStepSubmit} />}
+
+      {activeStep === 1 && (
+        <GithubConnectionStep formManager={githubConnection} onSubmit={onStepSubmit} />
+      )}
+
+      {activeStep === 2 && (
+        <TechnologiesOverviewStep formManager={technologiesOverview} onSubmit={onStepSubmit} />
+      )}
```

Comment:
Ok this one is pretty complicated for me, could we have a call in some time so you can explain it for me? Because I dont understand everything :D